spynnaker/pyNN/models/neuron/synapse_dynamics/synapse_dynamics_structural.py

```diff
@@ -0,0 +1,567 @@
+from six import itervalues
+import numpy as np
+import collections
+
+from spinn_utilities.overrides import overrides
+from data_specification.enums.data_type import DataType
+from spynnaker.pyNN.models.neural_projections import ProjectionApplicationEdge
+from spynnaker.pyNN.models.neural_projections import ProjectionMachineEdge
+from .abstract_plastic_synapse_dynamics import AbstractPlasticSynapseDynamics
+from .abstract_synapse_dynamics_structural import \
+    AbstractSynapseDynamicsStructural
+from .synapse_dynamics_stdp import SynapseDynamicsSTDP
+from .synapse_dynamics_static import SynapseDynamicsStatic
+from spynnaker.pyNN.utilities import constants
+
+
+class SynapseDynamicsStructural(AbstractSynapseDynamicsStructural):
+    __slots__ = [
+        "_f_rew",
+        "_p_rew",
+        "_weight",
+        "_delay",
+        "_s_max",
+        "_lateral_inhibition",
+        "_sigma_form_forward",
+        "_sigma_form_lateral",
+        "_p_form_forward",
+        "_p_form_lateral",
+        "_p_elim_dep",
+        "_p_elim_pot",
+        "_grid",
+        "_random_partner",
+        "_connections",
+        "_actual_row_max_length"]
+
+    def __init__(self, stdp_model=None, f_rew=10 ** 4, weight=0, delay=1,
+                 s_max=32, sigma_form_forward=2.5, sigma_form_lateral=1,
+                 p_form_forward=0.16, p_form_lateral=1,
+                 p_elim_dep=0.0245, p_elim_pot=1.36 * 10 ** -4,
+                 grid=np.array([16, 16]), lateral_inhibition=0,
+                 random_partner=False,
+                 seed=None):
+
+        AbstractSynapseDynamicsStructural.__init__(self)
+        self._f_rew = f_rew  # Hz
+        self._p_rew = 1. / self._f_rew  # ms
+        self._weight = weight
+        self._delay = delay
+        self._s_max = s_max  # maximum number of presynaptic neurons
+        self._lateral_inhibition = lateral_inhibition
+        self._sigma_form_forward = sigma_form_forward
+        self._sigma_form_lateral = sigma_form_lateral
+        self._p_form_forward = p_form_forward
+        self._p_form_lateral = p_form_lateral
+        self._p_elim_dep = p_elim_dep
+        self._p_elim_pot = p_elim_pot
+        self._grid = np.asarray(grid, dtype=int)
+        self._random_partner = random_partner
+        self._connections = {}
+
+        self.fudge_factor = 1.5
+        self._actual_row_max_length = self._s_max
+
+        if stdp_model is not None and \
+                isinstance(stdp_model, SynapseDynamicsSTDP):
+            self._super = SynapseDynamicsSTDP(
```

Comment:
REALLY REALLY DONT LIKE THAT! change to a more understandable name. i had freaks out that you were changing your inhertance there. 

you've also broken pycharm's checking with this class, so im assuming its this param thats messign everything up.

Reply:
this is everywhere in this class. fix

Reply:
if your trying to fake the inhirtance hiarky, just make versions of this with correct inhirtance